aten/src/ATen/native/cuda/MiscUtils.h

```diff
@@ -49,6 +53,25 @@ static inline magma_int_t magma_int_cast(int64_t value, const char* varname) {
   }
   return result;
 }
+
+// MAGMA functions that don't take a magma_queue_t aren't stream safe
+// Work around this by synchronizing with the default stream
+struct MagmaStreamSyncGuard {
+  MagmaStreamSyncGuard() {
+    auto stream = at::cuda::getCurrentCUDAStream();
+    if (stream != 0) {
+      AT_CUDA_CHECK(cudaStreamSynchronize(stream));
+    }
+  }
+
+  ~MagmaStreamSyncGuard() noexcept(false) {
+    if (at::cuda::getCurrentCUDAStream() == 0) {
+      AT_CUDA_CHECK(cudaGetLastError());
+    } else {
+      AT_CUDA_CHECK(cudaStreamSynchronize(0));
```

Comment:
If the error returned by magma kernels was non-sticky, it won't be checked (because cudaStreamSynchronize will likely succeed). Also, you can use getDefaultCUDAStream here instead of 0 here and above?